swift/ql/test/query-tests/Security/CWE-757/TLS13IsNotUsed.swift

```diff
@@ -0,0 +1,173 @@
+}
+
+func case_2() {
+  let config = URLSessionConfiguration()
+  config.tlsMinimumSupportedProtocolVersion = tls_protocol_version_t.TLSv10 // BAD
+}
+
+func case_3() {
+  let config = URLSessionConfiguration()
+  config.tlsMinimumSupportedProtocolVersion = tls_protocol_version_t.TLSv11 // BAD
+}
+
+/// tlsMaximumSupportedProtocolVersion
+
+func case_4() {
+  let config = URLSessionConfiguration()
+  config.tlsMaximumSupportedProtocolVersion = tls_protocol_version_t.TLSv12 // GOOD
+}
+
+func case_5() {
+  let config = URLSessionConfiguration()
+  config.tlsMaximumSupportedProtocolVersion = tls_protocol_version_t.TLSv10 // BAD
+}
+
+/// tlsMinimumSupportedProtocol
+
+func case_6() {
+  let config = URLSessionConfiguration()
+  config.tlsMinimumSupportedProtocol = SSLProtocol.tlsProtocol10 // BAD
+}
+
+func case_7() {
+  let config = URLSessionConfiguration()
+  config.tlsMinimumSupportedProtocol = SSLProtocol.tlsProtocol12 // GOOD
+}
+
+/// tlsMaximumSupportedProtocol
+
+func case_8() {
+  let config = URLSessionConfiguration()
+  config.tlsMaximumSupportedProtocol = SSLProtocol.tlsProtocol10 // BAD
+}
+
+func case_9() {
+  let config = URLSessionConfiguration()
+  config.tlsMaximumSupportedProtocol = SSLProtocol.tlsProtocol12 // GOOD
+}
+
+/// Indirect assignment (global vars)
+
+let badGlobalVersion = tls_protocol_version_t.TLSv10
+let goodGlobalVersion = tls_protocol_version_t.TLSv12
+
+func case_10() {
+  let config = URLSessionConfiguration()
+  config.tlsMinimumSupportedProtocolVersion = badGlobalVersion // BAD [not detected]
+}
+
+func case_11() {
+  let config = URLSessionConfiguration()
+  config.tlsMinimumSupportedProtocolVersion = goodGlobalVersion // GOOD
+}
+
+/// Indirect assignment (function calls)
+
+func getBadTLSVersion() -> tls_protocol_version_t {
+  return tls_protocol_version_t.TLSv10
+}
+
+func getGoodTLSVersion() -> tls_protocol_version_t {
+  return tls_protocol_version_t.TLSv13
+}
+
+func case_12() {
+  let config = URLSessionConfiguration()
+  config.tlsMinimumSupportedProtocolVersion = getBadTLSVersion() // BAD
+}
+
+func case_13() {
+  let config = URLSessionConfiguration()
+  config.tlsMinimumSupportedProtocolVersion = getGoodTLSVersion() // GOOD
+}
+
+/// Indirect assignment (via call arguments)
+
+func setTLSVersion(_ config: URLSessionConfiguration, _ version: tls_protocol_version_t) {
+  config.tlsMinimumSupportedProtocolVersion = version
+}
+
+func case_14() {
+  let config = URLSessionConfiguration()
+  setTLSVersion(config, tls_protocol_version_t.TLSv11) // BAD
+}
+
+func case_15() {
+  let config = URLSessionConfiguration()
+  setTLSVersion(config, tls_protocol_version_t.TLSv13) // GOOD
+}
+
+/// Indirect assignment (via external entity)
+
+struct BadDefault {
+  let TLSVersion: tls_protocol_version_t = tls_protocol_version_t.TLSv11
+}
+
+func case_16() {
+  let def = BadDefault()
+  let config = URLSessionConfiguration()
+  config.tlsMinimumSupportedProtocolVersion = def.TLSVersion // BAD [not detected]
```

Comment:
Yeah, I think this makes sense as well. I don't think we recognize the assignment from line 138 as an SSA definition.

Thanks for some awesome tests!